WeatherApp/WeatherApp/App/Data/Services/DataService.swift

```diff
@@ -0,0 +1,36 @@
+import Foundation
+
+protocol DataServiceProtocol {
+
+    func storeCities(cities: [City])
+    func getCities() -> [City]
+
+}
+
+class DataService: DataServiceProtocol {
+
+    private let citiesKey = "savedCities"
+
+    func storeCities(cities: [City]) {
+        let encoder = JSONEncoder()
+        do {
+            let data = try encoder.encode(cities)
```

Comment:
this is good, I just find it neater to instantiate the encoder and decoder insntly if it will not be used anywhere else

`try JSONEncoder().encode(cities)` same with the `getCities`